crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -115,13 +115,12 @@ impl<'a> Checker<'a> {
 }
 
 impl<'a> Checker<'a> {
-    /// Return `true` if a patch should be generated under the given autofix
-    /// `Mode`.
+    /// Return `true` if a patch should be generated for a given [`Rule`].
     pub fn patch(&self, code: Rule) -> bool {
-        self.settings.rules.should_fix(code)
+        self.settings.rules.should_fix(code) && !self.ctx.in_nested_f_string()
```

Comment:
Adding additional logic here is problematic because we might now no-longer generate fixes for rules that are otherwise marked as `AutofixKind::Always`.

Reply:
Yeah. I'm not sure how else to enforce this though. Any ideas? :/

Reply:
Not really. The most ideal solution in my view would be to automatically downgrade all fixes to `Manual` but this is hard today because ruff has no centralized logic collecting all diagnostics (every rule has the liberty to manipulate the `diagnostics` vec).